streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java

```diff
@@ -534,11 +543,13 @@ public void shouldFetchAndIterateOverKeyRange() {
         );
     }
 
+    @SuppressWarnings("deprecation")
```

Comment:
This test actually exercises the deprecated method. So, perhaps we should just add a justifying comment, as I proposed above.